packages/bsky/src/app-migrations/backfill.ts

```diff
@@ -0,0 +1,190 @@
+}
+
+const indexStreamCursor = async (ctx: AppContext, pdsDb: PdsDatabase) => {
+  const lastSeq = await pdsDb.db
+    .selectFrom('repo_seq')
+    .select('seq')
+    .where('seq', 'is not', null)
+    .orderBy('seq', 'desc')
+    .limit(1)
+    .executeTakeFirst()
+
+  if (!ctx.cfg.repoProvider || !lastSeq) {
+    throw new Error('could not index stream')
+  }
+
+  await ctx.db.db
+    .insertInto('subscription')
+    .values({
+      service: ctx.cfg.repoProvider,
+      method: 'com.atproto.sync.subscribeRepos',
+      state: JSON.stringify({ cursor: lastSeq }),
+    })
+    .execute()
+}
+
+const indexAccounts = async (ctx: AppContext, pdsDb: PdsDatabase) => {
+  // @TODO should we copy over user_account.createdAt for indexedAt time?
+  const now = new Date().toISOString()
+
+  let lastDid: string | undefined
+  /* eslint-disable */
+  while (true) {
+    let builder = pdsDb.db
+      .selectFrom('did_handle')
+      .innerJoin('repo_root', 'repo_root.did', 'did_handle.did')
+      .innerJoin('ipld_block', (join) =>
+        join
+          .onRef('ipld_block.creator', '=', 'repo_root.did')
+          .onRef('ipld_block.cid', '=', 'repo_root.root'),
+      )
+      .selectAll()
+      .orderBy('did_handle.did', 'asc')
+      .limit(PAGINATION_SIZE)
+    if (lastDid) {
+      builder = builder.where('did', '>', lastDid)
+    }
+    const page = await builder.execute()
+
+    const actorVals = page.map((user) => ({
+      did: user.did,
+      handle: user.handle,
+      indexedAt: now,
+      takedownId: user.takedownId,
+    }))
+    await ctx.db.db.insertInto('actor').values(actorVals).execute()
+
+    const actorSyncVals = page.map((user) => {
+      const root = def.commit.schema.parse(cborDecode(user.content))
+      return {
+        did: user.did,
+        commitCid: user.cid,
+        commitDataCid: root.data.toString(),
+        rebaseCount: 0,
+        tooBigCount: 0,
+      }
+    })
+    await ctx.db.db.insertInto('actor_sync').values(actorSyncVals).execute()
+
+    lastDid = page.at(-1)?.did
+    if (!lastDid) break
+  }
+}
+
+const indexRecords = async (ctx: AppContext, pdsDb: PdsDatabase) => {
+  const indexingSrvc = ctx.services.indexing(ctx.db)
+
+  let lastDid: string | undefined
+  let lastCid: string | undefined
+  /* eslint-disable */
+  while (true) {
+    let builder = pdsDb.db
+      .selectFrom('record')
+      .innerJoin('ipld_block', (join) =>
+        join
+          .onRef('record.did', '=', 'ipld_block.creator')
+          .onRef('record.cid', '=', 'ipld_block.cid'),
+      )
+      .selectAll()
+      .orderBy('record.did', 'asc')
+      .orderBy('record.cid', 'asc')
+      .limit(PAGINATION_SIZE)
+
+    if (lastDid && lastCid) {
+      builder = builder
+        .where('record.did', '>', lastDid)
+        .orWhere((clause) =>
+          clause
+            .where('record.did', '=', lastDid ?? '')
+            .where('record.cid', '>', lastCid ?? ''),
+        )
```

Comment:
I think it may be worthwhile to switch this to a row value comparison similar to #1174 so we're sure to hit the `(did, cid)` index for pagination.